resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -792,12 +795,41 @@ private void onExpectedResponse(
                 } while (resolved != null);
 
                 if (resolved == null) {
-                    continue;
+                    assert questionName.isEmpty() || questionName.charAt(questionName.length() - 1) == '.';
+
+                    for (String searchDomain : parent.searchDomains()) {
+                        if (searchDomain.isEmpty()) {
+                            continue;
+                        }
+
+                        final String fqdn;
+                        if (searchDomain.charAt(searchDomain.length() - 1) == '.') {
+                            fqdn = questionName + searchDomain;
+                        } else {
+                            fqdn = questionName + searchDomain + '.';
+                        }
+                        if (recordName.equals(fqdn)) {
+                            resolved = recordName;
+                            break;
+                        }
+                    }
+                    if (resolved == null) {
+                        if (logger.isInfoEnabled()) {
+                            logger.info("Ignoring record {} as it contains a different name than the " +
```

Comment:
not sure if info is really the right level here.. maybe debug ?

Reply:
I'm fine to start with `debug` and increase level later if necessary